bwapi/Library/BWAPILIBUnitTest/UnitTypeTest.cpp

```diff
@@ -14,866 +17,1052 @@
 using namespace BWAPI;
 
 std::string get_unitType_string(const testing::TestParamInfo<UnitType>& value) {
-  std::string result = value.param.getName();
-  if (result.empty()) {
-    result = std::to_string(value.index);
-  }
-  return result;
+	std::string result = value.param.getName();
+	if (result.empty()) {
+		result = std::to_string(value.index);
+	}
+	return result;
 }
 
 TEST_P(UnitTypeFixture, maxHitPoints) {
-  UnitType u = GetParam();
-  ASSERT_EQ(u.maxHitPoints(), BW::units.hitPoints[u.getID()] / 256);
+	UnitType u = GetParam();
+	ASSERT_EQ(u.maxHitPoints(), BW::units.hitPoints[u.getID()] / 256);
 }
 
 TEST_P(UnitTypeFixture, maxShields) {
-  UnitType u = GetParam();
-  ASSERT_EQ(u.maxShields(), BW::units.shieldEnable[u.getID()] ? BW::units.shields[u.getID()] : 0);
+	UnitType u = GetParam();
+	ASSERT_EQ(u.maxShields(), BW::units.shieldEnable[u.getID()] ? BW::units.shields[u.getID()] : 0);
 }
 
+TEST_P(UnitTypeFixture, airWeapon) {
+	UnitType u = GetParam();
+	if (u == UnitTypes::Terran_Goliath) {
+		ASSERT_EQ(u.airWeapon().getID(), 8);
```

Comment:
Use `BWAPI::Weapons::Enum` instead of an integer value. Same with everything else that uses raw values.

Reply:
Also for this it might be better to use SC logic -> get subunit type, if the subunit is not None then use that type instead to check.

Reply:
i.e.

```cpp
if (BW::units.subunit[u.getID()] != UnitTypes::Enum::None)
  u = BW::units.subunit[u.getID()];
```